src/crypto/rsa_provider/rsa_provider_impl.cpp

```diff
@@ -157,7 +157,7 @@ namespace libp2p::crypto::rsa {
       const Signature &signature,
       const PublicKey &public_key) const {
     OUTCOME_TRY(x509_key, RsaProviderImpl::getPublicKeyFromBytes(public_key));
-    EVP_PKEY *key = X509_PUBKEY_get0(x509_key.get());
+    EVP_PKEY *key = X509_PUBKEY_get(x509_key.get());
```

Comment:
Please check Openssl docs:
Key returned by `X509_PUBKEY_get` should be freed using `EVP_PKEY_free `. Otherwise memory might be leaked.

Alternatively, we can just keep using `EVP_PKEY_get0`

Reply:
Fixed